lib/evmone/baseline.cpp

```diff
@@ -356,6 +373,20 @@ evmc_result execute(const VM& vm, ExecutionState& state, const CodeAnalysis& ana
 evmc_result execute(evmc_vm* c_vm, const evmc_host_interface* host, evmc_host_context* ctx,
     evmc_revision rev, const evmc_message* msg, const uint8_t* code, size_t code_size) noexcept
 {
+    if (rev >= EVMC_CANCUN && is_eof_container({code, code_size}))
+    {
+        // TODO(EOF): The initcode must be validated. Doing this just before execution is
+        //   good because validation can be combined with analysis/loading. But consider also
+        //   other places like create instructions.
+        if (const auto err = validate_eof(rev, {code, code_size});
+            err != EOFValidationError::success)
+        {
+            // TODO(EOF): This should never happen but protects against invalid tests for now.
+            std::cerr << "EOF validation failed: " << get_error_message(err) << '\n';
```

Comment:
Are we sure we want to keep `std::cerr` in the library portion?

Reply:
As the comment says, this protects against misuse from CLI and tests. I think it has to stay for now until we add EOF validation in other places.